File: Arrays&Hashing/238ProductOfArrayExceptSelf.cpp
#include<vector>
class Solution {
public:
    std::vector<int> productExceptSelf(std::vector<int>& nums) {
        int pre = 1;
        std::vector<int> result;
        for(const auto& num : nums){
            result.push_back(pre);
            pre *= num;
        }
        int post = 1;
        for(int i = nums.size()-1; i >= 0; i--){
            result[i] *= post;
            post *= nums[i];
        }
        return result;
    }
};

File: Arrays&Hashing/238ProductOfArrayExceptSelf.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        answer = []
        pre = 1
        for i in range(len(nums)):
            answer.append(pre)
            pre = pre * nums[i]
        post = 1
        for i in range(len(nums)-1,-1,-1):
            answer[i] *= post
            post = post * nums[i]
        return answer
        
        